### DISCUSSION TITLE
Problem with OData filter on GET

### DISCUSSION THREAD

**Discussion by ideas2appsaustralia** [Azure/azure-mobile-apps#325] [Q&A] [answered]
Previously created the xamarin.forms app and ASP.Net backend using the eBook. Implemented MSAL and B2C auth and then changed to the Azure Mobile Apps for ASP.NET 6. for the backend which is published to an Azure Apps Service.

After finally getting B2C auth to work on the API now find that the GET returns an error when an OData filter is included (which was used in the original ASP.Net Framework used for the EntityTable used).

Using Postman and an access token I can GET all entities in a table called UserSurvey when I do not include the filter eg GET https://greenlightbackend.azurewebsites.net/tables/UserSurvey, however when using the $filter it errors.

The Get used is;
GET https://greenlightbackend.azurewebsites.net/tables/UserSurvey?$filter=(updatedAt ge datetimeoffset'1970-01-01T00:00:00.0000000%2B00:00')&$orderby=updatedAt&$skip=0&$top=50&__includeDeleted=true

The error is;
Response: BadRequest
message: "The query specified in the URI is not valid. Could not find a property named 'updatedAt' on type 'Microsoft.AspNetCore.Datasync.Models.PagedResult'.","exceptionMessage":"Could not find a property named 'updatedAt' on type 'Microsoft.AspNetCore.Datasync.Models.PagedResult'.","exceptionType":"Microsoft.OData.ODataException","stackTrace":"   at Microsoft.OData.UriParser.EndPathBinder.GeneratePropertyAccessQueryForOpenType(EndPathToken endPathToken, SingleValueNode parentNode)\r

Any ideas? Could the database now be incompatible? I ran add-migration initial and update (with my existing database already created though)

**Comment by adrianhall** [comment, marked as answer]
Can you post the UserSurvey model and controller?

**Reply by ideas2appsaustralia** [reply]
Adrian

UseSurveyController is
`using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Datasync;
using Microsoft.AspNetCore.Datasync.EFCore;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.OData.Query;
using Microsoft.Identity.Web.Resource;
using MobileBackendCore.Db;
using MobileBackendCore.Models;

namespace MobileBackendCore.Controllers
{
	[Authorize]
    [RequiredScope("tasks.read")]
    [EnableQuery(PageSize = 50)]
    [Route("tables/userSurvey")]

    public class UserSurveyController : TableController<UserSurvey>
    {
        public UserSurveyController(AppDbContext context, ILogger<UserSurveyController> logger)
            : base(new EntityTableRepository<UserSurvey>(context))
        {
            Logger = logger;
            Options = new TableControllerOptions
            {
                EnableSoftDelete = true,
                UnauthorizedStatusCode = StatusCodes.Status418ImATeapot
            };
        }

    }`

UserSurvey model is

`using Microsoft.AspNetCore.Datasync.EFCore;

namespace MobileBackendCore.Models
{
	public class UserSurvey : EntityTableData
    {
        public string UserId { get; set; }
        public string UserSurveyId { get; set; }
        public string SurveyResponseId { get; set; }
        public int? Choice { get; set; }
        public bool Completed { get; set; }
    }
}`

**Reply by ideas2appsaustralia** [reply]
Appears as any type of OData query doesnt work. For example

https://greenlightbackend.azurewebsites.net/tables/UserSurvey?$filter=(updatedAt ge datetimeoffset'1970-01-01T00:00:00.0000000%2B00:00')

    "message": "The query specified in the URI is not valid. Unrecognized 'Edm.String' literal 'datetimeoffset'1970-01-01T00:00:00.0000000+00:00'' at '14' in '(updatedAt ge datetimeoffset'1970-01-01T00:00:00.0000000+00:00')'.",
    "exceptionMessage": "Unrecognized 'Edm.String' literal 'datetimeoffset'1970-01-01T00:00:00.0000000+00:00'' at '14' in '(updatedAt ge datetimeoffset'1970-01-01T00:00:00.0000000+00:00')'.",
    "exceptionType": "Microsoft.OData.ODataException",

or

https://greenlightbackend.azurewebsites.net/tables/UserSurvey?$skip=0&$top=50

"message": "The query specified in the URI is not valid. The requested resource is not a collection. Query options $filter, $orderby, $count, $skip, and $top can be applied only on collections.",
    "exceptionMessage": "The requested resource is not a collection. Query options $filter, $orderby, $count, $skip, and $top can be applied only on collections.",
    "exceptionType": "Microsoft.OData.ODataException",

**Reply by adrianhall** [reply]
Remove the [EnableQuery] - it's interfering with the tablecontroller definition.

**Reply by ideas2appsaustralia** [reply]
that worked Adrian! thanks as I don't know how long that would have taken me to work out
So now the xamarin.forms mobile Client App authenticating via B2C MSAL with a native app and web api app in the B2C tenant and now ASP.NET core API on Azure App Service with Azure SQL Database are all talking to each other.

One more (huge) step forward!

Looking forward to your updated docs and more on ASP.NET core and Azure Mobil Apps

thanks
Scott
